cmd/collector/app/metrics.go

```diff
@@ -169,19 +166,27 @@ func (m metricsBySvc) countTracesByServiceName(serviceName string) {
 // total number of stored counters, so if it exceeds say the 90% threshold
 // an alert should be raised to investigate what's causing so many unique
 // service names.
-func (m *countsBySvc) countByServiceName(serviceName string) {
+func (m *countsBySvc) countByServiceName(serviceName string, isDebug bool) {
 	serviceName = NormalizeServiceName(serviceName)
+	if isDebug {
+		m.x(serviceName, "true", m.debugCounts)
+	} else {
+		m.x(serviceName, "false", m.counts)
+	}
+}
+
+func (m *countsBySvc) x(serviceName, debugStr string, counts map[string]metrics.Counter) {
```

Comment:
why do you need an extra function with such a great name? just set two local vars above and continue in the same function.

Reply:
duuude